api/src/main/java/org/pmiops/workbench/interceptors/ElapsedTimeDistributionInterceptor.java

```diff
@@ -0,0 +1,70 @@
+package org.pmiops.workbench.interceptors;
+
+import com.google.api.client.http.HttpMethods;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.pmiops.workbench.monitoring.LogsBasedMetricService;
+import org.pmiops.workbench.monitoring.MeasurementBundle;
+import org.pmiops.workbench.monitoring.labels.MetricLabel;
+import org.pmiops.workbench.monitoring.views.DistributionMetric;
+import org.springframework.stereotype.Service;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+@Service
+public class ElapsedTimeDistributionInterceptor extends HandlerInterceptorAdapter {
```

Comment:
TOL: Since this is about tracking time, it could make sense to fold this and tracing together into a TimeTrackingInterceptor...